### PR TITLE
Test case/manual/jack detection

### PR DESCRIPTION
**Pull request by stripes416**
Adding manual walkthrough instructions for the current three jack detection tests.

### CODE REVIEW

File: test-case/manual/jack_detection_boot.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Jack detection for headset/HDMI/DP during boot test
+Jack detection for headset/HDMI/DP during boot is part of a series of manual
+tests.
+
+# Preconditions
+1. System is powered off.
+
+# Test Description
+* Plug headset into unit, and determine if status is on as expected after power
+on and off after power down and removal.
```
Comment (xiulipan) [inline test-case/manual/jack_detection_boot.md]: What do you mean here by 
```
after power on and off after power down and removal.
```

I think we can only check the Jack status after power on?

File: test-case/manual/jack_detection_boot.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* Repeat for both HDMI and DisplayPort.
```
Comment (xiulipan) [inline test-case/manual/jack_detection_boot.md]: What if there is no HDMI/DP port? Maybe add something like ` if exists`

File: test-case/manual/jack_detection_boot.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## Headset
+1. Plug headset into 3 1/2 mm jack.
```
Comment (xiulipan) [inline test-case/manual/jack_detection_boot.md]:
```suggestion
1. Plug headset into 3.5 mm jack.
```
But I think we also have some 2.5 mm jack and type-c jack?

File: test-case/manual/jack_detection_boot.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+2. Power on system.
+3. Check amixer contents via:
+    ```
+    amixer contents
```
Comment (xiulipan) [inline test-case/manual/jack_detection_boot.md]:
```suggestion
    amixer contents | grep -i jack -A 3
```

File: test-case/manual/jack_detection_idle.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+# Test Description
+* Plug headset into unit, and determine if status is updated as expected.
+* Repeat for both HDMI and DisplayPort.
+
+## Headset
+1. Plug headset into 3 1/2 mm jack.
+2. Check amixer contents via:
+    ```
+    amixer contents
```
Comment (xiulipan) [inline test-case/manual/jack_detection_idle.md]:
```suggestion
    amixer contents | grep -i jack -A 3
```

File: test-case/manual/jack_detection_idle.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+# Preconditions
+1. Runtime PM status is suspended, no playback or capture process is running.
+
+# Test Description
+* Plug headset into unit, and determine if status is updated as expected.
+* Repeat for both HDMI and DisplayPort.
+
+## Headset
+1. Plug headset into 3 1/2 mm jack.
```
Comment (xiulipan) [inline test-case/manual/jack_detection_idle.md]:
```suggestion
1. Plug headset into 3.5 mm jack.
```

And if the test is for idle? Maybe we should check if system is in suspend first?

File: test-case/manual/jack_detection_playback_capture.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+status is updated as expected.
+* Repeat for both HDMI and DisplayPort.
+
+## Headset
+1. In terminal 1: run aplay/arecord via:
+    ```
+    aplay -Dhw:0,0 -r 48000 -c 2 -f s16_le -d 60 /dev/zero
+    arecord -Dhw:0,0 -r 48000 -c 2 -f s16_le -d 60 /dev/null
+    ```
+2. Plug headset into 3 1/2 mm jack.
```
Comment (xiulipan) [inline test-case/manual/jack_detection_playback_capture.md]:
```suggestion
2. Plug headset into 3.5 mm jack.
```

File: test-case/manual/jack_detection_playback_capture.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+3. In terminal 2: check amixer contents via:
+    ```
+    amixer contents
```
Comment (xiulipan) [inline test-case/manual/jack_detection_playback_capture.md]:
```suggestion
    amixer contents | grep -i jack -A 3
```

File: test-case/manual/jack_detection_boot.md (hunk 1 above)
Comment (stripes416) [reply test-case/manual/jack_detection_boot.md]: Ha yeah, I definitely worded that weird.

File: test-case/manual/jack_detection_boot.md (hunk 4 above)
Comment (stripes416) [reply test-case/manual/jack_detection_boot.md]: I've been playing with ways to make this easier on the user (me). Its fairly annoying to have to rerun the script every time I plug / unplug & I was looking for a way to have the plug event update as it happens, without me re running commands.

This seems to do exactly what I want. What do you think @xiulipan ?

`watch -d -n 1 "amixer contents | grep -i jack -A 2"`

File: test-case/manual/jack_detection_idle.md (hunk 5 above)
Comment (stripes416) [reply test-case/manual/jack_detection_idle.md]: https://github.com/thesofproject/sof-test/pull/44#discussion_r368173834

File: test-case/manual/jack_detection_playback_capture.md (hunk 8 above)
Comment (stripes416) [reply test-case/manual/jack_detection_playback_capture.md]: https://github.com/thesofproject/sof-test/pull/44#discussion_r368173834

File: test-case/manual/jack_detection_boot.md (hunk 4 above)
Comment (stripes416) [reply test-case/manual/jack_detection_boot.md]: Re-reading this, I'd like to clarify, the `watch -d ....` really only applies to the other two jack detection tests, not really the (this) boot test.

File: test-case/manual/jack_detection_idle.md (hunk 6 above)
Comment (stripes416) [reply test-case/manual/jack_detection_idle.md]: > And if the test is for idle? Maybe we should check if system is in suspend first?

Thats listed as one of the preconditions of the test.

File: test-case/manual/jack_detection_boot.md (hunk 4 above)
Comment (xiulipan) [reply test-case/manual/jack_detection_boot.md]: @stripes416 You can come with advanced CMD to save time.
They will to be base for the automation test later.

File: test-case/manual/jack_detection_boot.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    * Status is **OFF** if removed during power down cycle
+* No unexpected errors should be present in dmesg during or after test
+  completion.
+
+### Notes
+* The various jacks (headset/HDMI/DP) are all named differently on each platform,
+    so there is no universal "data value to look for when testing headset". The
+    best way to determine where you need to look is to run:
+    ```
+    watch -d -n 1 "amixer contents | grep -i jack -A 2"
```
Comment (xiulipan) [inline test-case/manual/jack_detection_boot.md]: As you have mentioned, do we really need to use `watch `here?

Comment (stripes416) [reply test-case/manual/jack_detection_boot.md]: This is instructions for the user to figure out which jack they are supposed to be looking for changes on --- Not the general run instructions.

**Comment by fredoh9** [comment]
LGTM

**Pull request merged**